portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -35,17 +36,32 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private static final int REQUESTED_COMMENTS_LIMIT = 1000;
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Sort comments by date from newest to oldest
+    int maxCommentsNumber;
+
+    // Try to convert the value from the 'max-comments' field of the query string to int
+    try {
+      maxCommentsNumber = Integer.parseInt(request.getParameter("max-comments"));
+    } catch (NumberFormatException e) {
+      // Handle the exception
+      maxCommentsNumber = REQUESTED_COMMENTS_LIMIT;
+    }
+
+    // Assure that maxCommentsNumber is not negative
+    maxCommentsNumber = Math.max(maxCommentsNumber, 0);
```

Comment:
If you get a negative number I would fallback to the maximum instead. Returning 0 comments seems not very useful.

Reply:
Indeed. Done!